### PR TITLE
chore: moved linter config into .golangci.yaml for more control

### PR DESCRIPTION
**Pull request by garethjevans**
#### Submitter checklist

- [x] Change is code complete and matches issue description.
- [x] Change is covered by existing or new tests.

### CODE REVIEW

**Comment by garethjevans** [comment]
I'd like to get the gosec linter enabled on this one

**Comment by garethjevans** [comment]
fixes https://github.com/jenkins-x/jx/issues/7197

**Comment by hferentschik** [comment]
@garethjevans, so this is still WIP, right?

**Comment by garethjevans** [comment]
@hferentschik yeah, I'm going to try and get the gosec stuff enabled today

**Comment by garethjevans** [comment]
@hferentschik I think this is good to go now

File: .golangci.yml [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    - errcheck
+#    - funlen
+#    - goconst
+    - gofmt
+    - goimports
+    - goprintffuncname
+    - gosec
+#    - gosimple
+#    - ineffassign
+    - interfacer
```
Comment (ankitm123) [inline .golangci.yml:83]: Interfacer is deprecated, and not recommended for real code anymore, so may be it should not be enabled? https://github.com/mvdan/interfacer

File: hack/linter.sh [hunk 2]
```diff
@@ -11,36 +11,11 @@ DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 
 if ! [ -x "$(command -v golangci-lint)" ]; then
 	echo "Installing GolangCI-Lint"
-	${DIR}/install_golint.sh -b $GOPATH/bin v1.20.0
+	${DIR}/install_golint.sh -b $GOPATH/bin v1.27.0
```
Comment (ankitm123) [inline hack/linter.sh:14]: :100:

File: .golangci.yml (hunk 1 above)
Comment (garethjevans) [reply .golangci.yml:83]: Interesting, I didn't realise this.  I can remove in another PR (this was already enabled on the existing linter config)

**Pull request merged**